common/src/main/java/io/netty/util/internal/Cleaner0.java

```diff
@@ -72,31 +63,28 @@
         }
         CLEANER_FIELD_OFFSET = fieldOffset;
         CLEAN_METHOD = clean;
-        CLEANER_IS_RUNNABLE = cleanerIsRunnable;
 
         // free buffer if possible
-        freeDirectBuffer(direct);
+        freeDirectBuffer0(direct);
+    }
+
+    @Override
+    public void freeDirectBuffer(ByteBuffer buffer) {
+        freeDirectBuffer0(buffer);
     }
 
-    static void freeDirectBuffer(ByteBuffer buffer) {
+    private static void freeDirectBuffer0(ByteBuffer buffer) {
```

Comment:
I think we can remove this method, and just move the code into `freeDirectBuffer`. We invoke the cleaner above `clean.invoke(cleaner);` so we may be double freeing this buffer